www_src/pages/page/flattening.js

```diff
@@ -0,0 +1,16 @@
+var types = require('../../components/basic-element/basic-element.jsx').types;
```

Comment:
that way they can be tested and modified along with their corresponding instance methods